dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/aggregates/AclStore.java

```diff
@@ -28,16 +28,101 @@
 package org.hisp.dhis.tracker.export.trackedentity.aggregates;
 
 import java.util.List;
+import org.hisp.dhis.common.collection.CollectionUtils;
+import org.hisp.dhis.hibernate.jsonb.type.JsonbFunctions;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
 
 /**
  * @author Luciano Fiandesio
  */
-public interface AclStore {
-  List<Long> getAccessibleTrackedEntityTypes(String userUID, List<String> userGroupUIDs);
+@Repository("org.hisp.dhis.tracker.trackedentity.aggregates.AclStore")
+class AclStore {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
 
-  List<Long> getAccessiblePrograms(String userUID, List<String> userGroupUIDs);
+  private static final String USER_SQL_PARAM_NAME = "userId";
 
-  List<Long> getAccessibleProgramStages(String userUID, List<String> userGroupUIDs);
+  private static final String USER_GROUP_SQL_PARAM_NAME = "userGroupUIDs";
 
-  List<Long> getAccessibleRelationshipTypes(String userUID, List<String> userGroupUIDs);
+  private static final String PUBLIC_ACCESS_CONDITION =
+      "sharing->>'public' LIKE '__r%' OR sharing->>'public' IS NULL";
```

Comment:
Does this mean the functionally it is the same if shating->>public begins with __r or null?